lib/rucio/core/rule_grouping.py

```diff
@@ -893,6 +916,7 @@ def __create_lock_and_replica(file, dataset, rule, rse_id, staging_area, availab
                                                                 adler32=file['adler32'],
                                                                 ds_scope=dataset['scope'],
                                                                 ds_name=dataset['name'],
+                                                                lifetime=maximum_pin_lifetime if staging_required else None,
```

Comment:
This is correct (The comment with the lifetime limit still holds though)
I am wondering though if it works. E.g. For a transfer from an RSE, somewhere in the world, to the MAS RSE, will the PIN lifetime by communicated? Or will this just be communicated for a STAGEIN?